server/app/api/v1/line.py

```
@@ -31,103 +32,130 @@
 ):
-    signature = request.headers['X-Line-Signature']
-
-    body = await request.body()
-    body = body.decode()
-
+    signature, body = await extract_line_request_data(request)
     agent_manager = get_agent_manager()
     try:
         agent_config = agent_manager.get_agent(agent_id)
     except KeyError:
         raise HTTPException(status_code=404, detail=f"Agent with ID '{agent_id}' not found.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)